src/devices/machine/s2350.cpp

```diff
@@ -0,0 +1,411 @@
+***************************************************************************/
+
+
+#include "emu.h"
+#include "s2350.h"
+
+// Shows register setup
+#define LOG_REG (1U << 1)
+// Show control lines
+#define LOG_LINE (1U << 2)
+// Function calls
+#define LOG_FUNC (1U << 3)
+// Sync detect
+#define LOG_SYNC (1U << 4)
+
+//#define VERBOSE (0xff)
+
+#include "logmacro.h"
+
+#define LOGREG(...)        LOGMASKED(LOG_REG, __VA_ARGS__)
+#define LOGLINE(...)       LOGMASKED(LOG_LINE, __VA_ARGS__)
+#define LOGFUNC(...)       LOGMASKED(LOG_FUNC, __VA_ARGS__)
+#define LOGSYNC(...)       LOGMASKED(LOG_SYNC, __VA_ARGS__)
+
+#ifdef _MSC_VER
+#define FUNCNAME __func__
+#else
+#define FUNCNAME __PRETTY_FUNCTION__
+#endif
+
+
+DEFINE_DEVICE_TYPE(S2350, s2350_device, "s2350", "American Microsystems Inc. S2350 USRT")
+
+
+s2350_device::s2350_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock) :
+	device_t(mconfig, S2350, tag, owner, clock),
+	m_tbmt(*this),
+	m_fct(*this),
+	m_rda(*this),
+	m_scr(*this),
+	m_ror(*this),
+	m_rpe(*this),
+	m_tx_cb(*this)
+{
+}
+
+void s2350_device::device_start()
+{
+	save_item(NAME(m_transmit_buffer_empty));
+	save_item(NAME(m_fill_char_transmitted));
+	save_item(NAME(m_transmitter_holding_reg));
+	save_item(NAME(m_transmitter_fill_reg));
+	save_item(NAME(m_transmitter_shift_reg));
+	save_item(NAME(m_serial_tx_state));
+
+	save_item(NAME(m_received_data_available));
+	save_item(NAME(m_receiver_overrun));
+	save_item(NAME(m_receiver_parity_error));
+	save_item(NAME(m_sync_character_received));
+
+	save_item(NAME(m_receiver_output_reg));
+	save_item(NAME(m_receiver_sync_reg));
+	save_item(NAME(m_receiver_shift_reg));
+	save_item(NAME(m_serial_rx_state));
+
+	save_item(NAME(m_in_sync));
+}
+
+void s2350_device::device_reset()
+{
+	// transmitter flags
+	set_transmit_buffer_empty(true);
+	set_fill_char_transmitted(false);
+
+	// transmitter registers
+	m_transmitter_shift_reg = 0xff;
+	m_transmitter_holding_reg = 0xff;
+	m_transmitter_fill_reg = 0x00;
+
+	m_serial_tx_state = 0;
+
+	set_received_data_available(false);
+	set_receiver_overrun(false);
+	set_receiver_parity_error(false);
+	set_sync_character_received(false);
+
+	m_sync_search_active = false;
+	m_in_sync = false;
+
+	m_serial_rx_state = 0;
+
+	// receiver registers
+	m_receiver_output_reg = 0x00;
+	m_receiver_sync_reg = 0x00;
+	m_receiver_shift_reg = 0x00;
+}
+
+void s2350_device::transmitter_holding_reg_w(u8 data)
+{
+	LOGREG("%s: 0x%02x\n", FUNCNAME, data);
+
+	m_transmitter_holding_reg = data;
+	set_transmit_buffer_empty(false);
+}
+
+void s2350_device::transmit_fill_reg_w(u8 data)
+{
+	LOGREG("%s: 0x%02x\n", FUNCNAME, data);
+
+	m_transmitter_fill_reg = data;
+}
+
+void s2350_device::receiver_sync_reg_w(u8 data)
+{
+	LOGREG("%s: 0x%02x\n", FUNCNAME, data);
+
+	m_receiver_sync_reg = data;
+}
+
+u8 s2350_device::receiver_sync_search()
+{
+	set_sync_character_received(false);
+	m_sync_search_active      = true;
```

Comment:
Can this line lose the extra spaces?  There are no other similar lines in this function for it to align with.